### DISCUSSION TITLE
Can i have options for smoothly fps ?

### DISCUSSION THREAD

**Discussion by copconlt** [hacksider/Deep-Live-Cam#727] [Q&A]
Hi , i use RTX 4090 for install but fps so slowly

**Comment by hacksider** [comment]
Are you using the cuda? With 4090 I am getting 20fps on my side which is gokd enough for any movies.

**Comment by rotatorotator** [comment]
I have also a 4090, what steps could we do to boost the fps? Any idea so far? I am happy to dig in and prepare a PR, but I have no idea where to start!

**Reply by hacksider** [reply]
I am still doing some planning on the code if it can subdivide the frames it is getting on the camera and multi thread it but as of now it really doesn't improve anything.

**Reply by rotatorotator** [reply]
Do you consider to buffer 2-3 frames and perform batch processing on the GPU on them? With 30 Fps, a buffer of 3 images would make 100ms delay, but might increase the speed by utilizing the GPU more.

**Reply by hacksider** [reply]
Delaying 3 frames and spitting it again and doing the same process will delay it even further. It really makes no sense if you do it.

**Comment by BG-Hanna** [comment]
> Hi , i use RTX 4090 for install but fps so slowly

hey @copconlt, 
try these fixes it might help you: 
•1: try Setting Preferred GPU to RTX 4090 ((NVIDIA Control Panel → Manage 3D Settings))
•2:Ensure you're running the latest version from NVIDIA
•3:check CPU bottlenecking: Enable Resizable BAR in BIOS for better GPU memory access
•4:try cleaning the fans, or improve airflow if GPU is overheating.
   Also buffering extra frames can add latency instead of improving FPS. If the application allows, multi-threading optimizations in the code could help, but identifying the real bottleneck is key first.
Let me know if this improves your FPS!
